### PR TITLE
NIAD-128: scaffold of close quarter notification handling

### PR DESCRIPTION
**Pull request by Alicja-Kainos**

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerService.java [hunk 1]
```diff
@@ -56,8 +57,10 @@ public void handleRegistration(InboundMeshMessage meshMessage) {
 
         Streams.zip(inboundStateRecords.stream(), supplierQueueDataToSend.stream(), Pair::of)
             .forEach(pair -> {
-                inboundGpSystemService.publishToSupplierQueue(pair.getRight());
-                inboundStateRepository.save(pair.getLeft());
+                if (!pair.getRight().getTransactionType().equals(ReferenceTransactionType.Inbound.CLOSE_QUARTER_NOTIFICATION)) {
+                    inboundGpSystemService.publishToSupplierQueue(pair.getRight());
+                    inboundStateRepository.save(pair.getLeft());
```
Comment (grzegorz-robak) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerService.java]: I think that inbound state repository should be updated

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerService.java [hunk 2]
```diff
@@ -56,8 +57,10 @@ public void handleRegistration(InboundMeshMessage meshMessage) {
 
         Streams.zip(inboundStateRecords.stream(), supplierQueueDataToSend.stream(), Pair::of)
             .forEach(pair -> {
-                inboundGpSystemService.publishToSupplierQueue(pair.getRight());
-                inboundStateRepository.save(pair.getLeft());
+                if (!pair.getRight().getTransactionType().equals(ReferenceTransactionType.Inbound.CLOSE_QUARTER_NOTIFICATION)) {
```
Comment (grzegorz-robak) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerService.java]: Extract this condition to method

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerService.java (hunk 1 above)
Comment (Alicja-Kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerService.java]: done

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerService.java (hunk 2 above)
Comment (Alicja-Kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerService.java]: done

File: src/test/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerServiceTest.java [hunk 3]
```diff
@@ -70,6 +70,9 @@
         .parse("199201141619", DateTimeFormatter.ofPattern("yyyyMMddHHmm").withZone(TimestampService.UKZone))
         .toInstant();
     private static final Instant MESSAGE_2_TRANSLATION_TIME = MESSAGE_1_TRANSLATION_TIME.plusSeconds(100);
+    private static final Instant CQN_MESSAGE_TRANSLATION_TIME = ZonedDateTime
```
Comment (grzegorz-robak) [inline src/test/java/uk/nhs/digital/nhsconnect/nhais/inbound/RegistrationConsumerServiceTest.java:73]: Why didn't you use one of already defined `Instant` objects?

**Review by grzegorz-robak — APPROVED** [review]

**Review by mattd-kainos — APPROVED** [review]

**Pull request merged**